### PR TITLE
initial commit for xenon mediation tests

### PR DESCRIPTION
**Pull request by timcadman**
Added tests for the mediation package. Test by checking the test script runs correctly.

Currently just running functions from the package and checking the correct class is returned. Let me know if you want anything more expansive!

### CODE REVIEW

**Review by marikaris — CHANGES_REQUESTED** [review]
Please add the install of the package to the `install_release_script_dependencies.R` as well:
`install_github("datashield/dsMediationClient")`

On my machine the mediation test seems to be failing:
```
  Aggregated (mediateDS("med.fit.1a", "out.fit.1a", "ethn3_m", "agebirth_m_y", ) [===>---]  50% / 0sError: There are some DataSHIELD errors, list them with datashield.errors()
Execution halted
```
The armadillo logs say this:
```
2024-02-20 08:38:42.774 [http-nio-8080-exec-2|4343F9D4] INFO  o.m.armadillo.audit.AuditLogger - AuditEvent [timestamp=2024-02-20T07:38:42.774280Z, principal=m.k.slofstra@umcg.nl, type=EXECUTE_FAILURE, data={expression=mediateDS("med.fit.1a", "out.fit.1a", "ethn3_m", "agebirth_m_y", 
    FALSE, 0.95, TRUE, 100, 123, "med.out.1a"), sessionId=2EB9D82DDFC8FF6AD0314F684343F9D4, message=No such DataSHIELD 'AGGREGATE' method with name: mediateDS, type=ExpressionException, roles=[ROLE_OSLEOTS8EY_RESEARCHER, ROLE_RXA385I5SX_RESEARCHER]}]
```
Any idea? I can run the code in my R studio as well, if you think that will give you insight into why they are failing.

File: scripts/release/release-test.R [hunk 1]
```diff
@@ -449,6 +451,98 @@ verify_ds_obtained_mean <- function(ds_mean, expected_mean, expected_valid_and_t
...
+
+if(med_class == "mediate"){
+  cli_alert_success("ds.mediate passed")
+} else{
+  cli_alert_danger("ds.mediate failed")
+  exit_test("ds.mediate did not return the expected class")
+
+}
+
+}
```
Comment (clemens-tolboom) [inline scripts/release/release-test.R:476]: Please indent the whole

File: scripts/release/release-test.R [hunk 2]
```diff
@@ -449,6 +451,98 @@ verify_ds_obtained_mean <- function(ds_mean, expected_mean, expected_valid_and_t
...
+
+if(med_class == "neModel"){
+  cli_alert_success("ds.neModel passed")
+} else{
+  cli_alert_danger("ds.neModel failed")
+  exit_test("ds.neModel did not return the expected class")
+
+}
+
+}
```
Comment (clemens-tolboom) [inline scripts/release/release-test.R:511]: indent

File: scripts/release/release-test.R [hunk 3]
```diff
@@ -449,6 +451,98 @@ verify_ds_obtained_mean <- function(ds_mean, expected_mean, expected_valid_and_t
...
+med_class <- ds.class("impData")
+
+if(identical(med_class$armadillo, c("data.frame", "expData", "impData"))){
+  cli_alert_success("ds.neImpute passed")
+} else{
+  cli_alert_danger("ds.neImpute failed")
+  exit_test("ds.neImpute did not return the expected class")
+}
+
+}
```
Comment (clemens-tolboom) [inline scripts/release/release-test.R:529]: indent

File: scripts/release/release-test.R [hunk 4]
```diff
@@ -449,6 +451,98 @@ verify_ds_obtained_mean <- function(ds_mean, expected_mean, expected_valid_and_t
...
+med_class <- class(lht.out.1b$armadillo)
+
+if(med_class == "summary.neLht"){
+  cli_alert_success("ds.neLht passed")
+} else{
+  cli_alert_danger("ds.neLht failed")
+  exit_test("ds.neLht did not return the expected class")
+}
+
+}
```
Comment (clemens-tolboom) [inline scripts/release/release-test.R:544]: Indent

File: scripts/release/release-test.R [hunk 5]
```diff
@@ -825,6 +919,12 @@ compare_list_values(hist$density, density)
 cli_alert_info("Validating histogram mids")
 compare_list_values(hist$mids, mids)
 
+verify_mediate_class()
+verify_ne_weight_class()
+verify_ne_model_class()
+verify_ne_imp_class()
+verify_ne_lht_class()
```
Comment (clemens-tolboom) [inline scripts/release/release-test.R:926]: Nice use of classes :)

**Review by marikaris — APPROVED** [review]

**Pull request merged**